Could not find artifact io.prometheus:client_java:jar:1.3.5

The artifact (https://central.sonatype.com/artifact/io.github.jeremylong/vulnz):

```
<groupId>io.github.jeremylong</groupId>
<artifactId>vulnz</artifactId> 
<version>7.2.0</version>
```

defines a dependency of type jar:

```
<dependency>
      <groupId>io.prometheus</groupId>
      <artifactId>client_java</artifactId>
      <version>1.3.5</version>
      <scope>runtime</scope>
  </dependency>

```
That is causing an error:
Could not find artifact io.prometheus:client_java:jar:1.3.5


Because <artifactId>client_java</artifactId> is of type pom in maven central:
https://central.sonatype.com/artifact/io.prometheus/client_java
It is a parent pom.

I get this error in a maven build. 

I just have a single dependency:

<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
  <modelVersion>4.0.0</modelVersion>
  <groupId>test</groupId>
  <artifactId>test</artifactId>
  <packaging>jar</packaging>
  <version>1.0.0-SNAPSHOT</version>

  <dependencies>
    <dependency>
      <groupId>io.github.jeremylong</groupId>
      <artifactId>vulnz</artifactId>
      <version>7.2.0</version>
    </dependency>
  </dependencies>
</project>


When running 

mvn clean install 

then I get this error:
...
Downloading from central: https://repo.maven.apache.org/maven2/io/prometheus/prometheus-metrics-tracer-common/1.3.5/prometheus-metrics-tracer-common-1.3.5.jar
Downloaded from central: https://repo.maven.apache.org/maven2/io/prometheus/prometheus-metrics-exposition-formats/1.3.5/prometheus-metrics-exposition-formats-1.3.5.jar (2.1 MB at 6.5 MB/s)
Downloading from central: https://repo.maven.apache.org/maven2/io/prometheus/prometheus-metrics-tracer-otel/1.3.5/prometheus-metrics-tracer-otel-1.3.5.jar
Downloaded from central: https://repo.maven.apache.org/maven2/io/prometheus/prometheus-metrics-core/1.3.5/prometheus-metrics-core-1.3.5.jar (95 kB at 287 kB/s)
Downloading from central: https://repo.maven.apache.org/maven2/io/prometheus/prometheus-metrics-tracer-otel-agent/1.3.5/prometheus-metrics-tracer-otel-agent-1.3.5.jar
Downloaded from central: https://repo.maven.apache.org/maven2/io/prometheus/prometheus-metrics-tracer-common/1.3.5/prometheus-metrics-tracer-common-1.3.5.jar (3.3 kB at 9.9 kB/s)
Downloaded from central: https://repo.maven.apache.org/maven2/io/prometheus/prometheus-metrics-tracer-initializer/1.3.5/prometheus-metrics-tracer-initializer-1.3.5.jar (4.0 kB at 12 kB/s)
Downloaded from central: https://repo.maven.apache.org/maven2/io/prometheus/prometheus-metrics-tracer-otel/1.3.5/prometheus-metrics-tracer-otel-1.3.5.jar (4.0 kB at 12 kB/s)
Downloaded from central: https://repo.maven.apache.org/maven2/io/prometheus/prometheus-metrics-instrumentation-jvm/1.3.5/prometheus-metrics-instrumentation-jvm-1.3.5.jar (53 kB at 154 kB/s)
Downloaded from central: https://repo.maven.apache.org/maven2/io/prometheus/prometheus-metrics-tracer-otel-agent/1.3.5/prometheus-metrics-tracer-otel-agent-1.3.5.jar (4.3 kB at 12 kB/s)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  9.528 s
[INFO] Finished at: 2025-01-20T21:11:49+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project test: Could not resolve dependencies for project test:test:jar:1.0.0-SNAPSHOT
[ERROR] dependency: io.prometheus:client_java:jar:1.3.5 (runtime)
[ERROR]         Could not find artifact io.prometheus:client_java:jar:1.3.5 in central (https://repo.maven.apache.org/maven2)


And indeed by looking at maven central
https://central.sonatype.com/artifact/io.prometheus/client_java
-> there is no jar available at this groupId/artifactId 
The artifact is of type pom.

--- a/vulnz/README.md
+++ b/vulnz/README.md
@@ -116,6 +116,7 @@ There are a couple of ENV vars
 - `MAX_RECORDS_PER_PAGE_ARG` Using max records per page
 - `METRICS_ENABLE` If is set to `true`, OpenMetrics data for the vulnz cli can be retrieved via the endpoint http://.../metrics
 - `METRICS_WRITE_INTERVAL` Sets the update interval for generating metrics, in milliseconds. Default: `5000`
+- `METRICS_WRITE_FORMAT` Sets the output format for the metrics. Either openmetrics or Prometheus format. Default: `openmetrics`
 
 ### Run
 

--- a/vulnz/build.gradle
+++ b/vulnz/build.gradle
@@ -30,7 +30,7 @@ dependencies {
     implementation 'jakarta.transaction:jakarta.transaction-api:2.0.1'
     implementation 'jakarta.xml.bind:jakarta.xml.bind-api:4.0.2'
     implementation 'com.sun.activation:jakarta.activation:2.0.1'
-    implementation 'io.prometheus:client_java:1.3.5'
+    implementation 'io.prometheus:prometheus-metrics-core:1.3.5'
     implementation 'io.prometheus:prometheus-metrics-exposition-formats:1.3.5'
     implementation 'io.prometheus:prometheus-metrics-instrumentation-jvm:1.3.5'
 

--- a/vulnz/src/main/java/io/github/jeremylong/vulnz/cli/monitoring/PrometheusFileWriter.java
+++ b/vulnz/src/main/java/io/github/jeremylong/vulnz/cli/monitoring/PrometheusFileWriter.java
@@ -17,12 +17,15 @@
 package io.github.jeremylong.vulnz.cli.monitoring;
 
 import io.github.jeremylong.vulnz.cli.commands.CveCommand;
+import io.prometheus.metrics.expositionformats.ExpositionFormatWriter;
 import io.prometheus.metrics.expositionformats.OpenMetricsTextFormatWriter;
+import io.prometheus.metrics.expositionformats.PrometheusTextFormatWriter;
 import io.prometheus.metrics.instrumentation.jvm.JvmMetrics;
 import io.prometheus.metrics.model.registry.PrometheusRegistry;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Component;
@@ -41,6 +44,9 @@ public class PrometheusFileWriter {
     @Autowired
     private CveCommand command;
 
+    @Value("${metrics.writer.format:openmetrics}")
+    private String metricsFormat;
+
     @PostConstruct
     public void init() {
         JvmMetrics.builder().register();
@@ -52,11 +58,18 @@ public class PrometheusFileWriter {
         if (directory != null) {
             final PrometheusRegistry defaultRegistry = PrometheusRegistry.defaultRegistry;
             try (FileOutputStream out = new FileOutputStream(new File(directory, "metrics"))) {
-                OpenMetricsTextFormatWriter writer = new OpenMetricsTextFormatWriter(true, true);
-                writer.write(out, defaultRegistry.scrape());
+                createWriter().write(out, defaultRegistry.scrape());
             } catch (IOException e) {
                 LOG.error("Error writing metrics", e);
             }
+        }
+    }
+
+    private ExpositionFormatWriter createWriter() {
+        if ("prometheus".equalsIgnoreCase(metricsFormat)) {
+            return new PrometheusTextFormatWriter(true);
+        } else {
+            return new OpenMetricsTextFormatWriter(true, true);
         }
     }
 }

--- a/vulnz/src/main/resources/application.properties
+++ b/vulnz/src/main/resources/application.properties
@@ -9,3 +9,4 @@ spring.main.log-startup-info=false
 application.version=@version@
 metrics.enable=false
 metrics.write.interval=5000
+metrics.writer.format=openmetrics

